aws/sdk/examples/ebs/src/bin/list-snapshots.rs

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_types::region::{self, ProvideRegion};
+use ec2::{Client, Config, Error, Region, PKG_VERSION};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+struct Opt {
+    /// The AWS Region.
+    #[structopt(short, long)]
+    region: Option<String>,
+
+    /// Whether to display additional information.
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+/// Displays some information about the EBS snapshots you own in the Region.
+/// # Arguments
+///
+/// * `[-r REGION]` - The Region in which the client is created.
+///    If not supplied, uses the value of the **AWS_REGION** environment variable.
+///    If the environment variable is not set, defaults to **us-west-2**.
+/// * `[-v]` - Whether to display additional information.
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    tracing_subscriber::fmt::init();
+
+    let Opt { region, verbose } = Opt::from_args();
+
+    let region_provider = region::ChainProvider::first_try(region.map(Region::new))
+        .or_default_provider()
+        .or_else(Region::new("us-west-2"));
+
+    println!();
+
+    if verbose {
+        println!("EC2 version: {}", PKG_VERSION);
+        println!(
+            "Region:      {}",
+            region_provider.region().unwrap().as_ref()
+        );
+
+        println!();
+    }
+
+    let config = Config::builder().region(region_provider).build();
+    let client = Client::from_conf(config);
+
+    // You can find snapshots for any account by replacing
+    // "self" with their account ID.
+    let resp = client.describe_snapshots().owner_ids("self").send().await?;
```

Comment:
is `self` a special account id?

Reply:
Yes, it means whoever is running the code's account ID. I'll add that explanation as a comment.